embulk-core/src/main/java/org/embulk/spi/PageReader.java

```diff
@@ -97,19 +98,42 @@ public String getString(int columnIndex) {
         return page.getStringReference(index);
     }
 
+    /**
+     * Returns a Timestamp value.
+     *
+     * @deprecated Use {@link #getTimestampInstant(Column)} instead.
+     */
+    @Deprecated
+    @SuppressWarnings("deprecation")  // https://github.com/embulk/embulk/issues/1292
     public Timestamp getTimestamp(Column column) {
         // TODO check type?
-        return getTimestamp(column.getIndex());
+        return Timestamp.ofInstant(this.getTimestampInstant(column.getIndex()));
     }
 
+    /**
+     * Returns a Timestamp value.
+     *
+     * @deprecated Use {@link #getTimestampInstant(int)} instead.
+     */
+    @Deprecated
+    @SuppressWarnings("deprecation")  // https://github.com/embulk/embulk/issues/1292
     public Timestamp getTimestamp(int columnIndex) {
+        return Timestamp.ofInstant(this.getTimestampInstant(columnIndex));
+    }
+
+    public Instant getTimestampInstant(final Column column) {
```

Comment:
Probably just `getInstant` name would be descriptive enough. What do you think?

Reply:
Yeah, I also thought about that, but its naming convention has used Embulk's type names, not Java type names here. (See `getJson` while its Java type is `org.msgpack.value.Value`.)

That was the reason why I kept the word `Timestamp` here, but I understand `getInstant` is simpler.

@trung-huynh Can we have your idea?

Reply:
Or we can have just `getTimestamp `  if following Embulk's type names

Reply:
Actually cannot because there already is `getTimestamp` to return `org.embulk.spi.time.Timestamp`, and that cannot be removed/replaced immediately because many existing plugins are calling that.